File: my-folder/1544-count-good-nodes-in-binary-tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func goodNodes(root *TreeNode) int {
    var dfs func(root *TreeNode, maxValue int) int
    dfs = func(root *TreeNode, maxValue int) int {
        res := 0
        if root == nil {
            return res
        }

        if root.Val >= maxValue {
            res = 1
        } else {
            res = 0
        }
        maxValue = max(maxValue, root.Val)

        return res + dfs(root.Left, maxValue) + dfs(root.Right, maxValue)
    }

    return dfs(root, root.Val)
}

File: my-folder/1544-count-good-nodes-in-binary-tree/solution.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function goodNodes(root: TreeNode | null): number {
    let goodnodes: number = 0

    let dfs = function(node: TreeNode | null, maxVal: number) {
        if (node === null) {
            return
        }

        if (node.val >= maxVal) {
            goodnodes++
        }

        dfs(node.left, Math.max(node.val, maxVal))
        dfs(node.right, Math.max(node.val, maxVal))

    }
    dfs(root, root.val)
    return goodnodes
};
